src/Sniffer.php

```diff
@@ -0,0 +1,281 @@
+    {
+        // MUST be false when not fixing, and MUST be true when fixing.
+        // This affects how codesniffer treats various CLI args, changes the output, and defines how some rules are actioned.
+        if (!defined('PHP_CODESNIFFER_CBF')) {
+            define('PHP_CODESNIFFER_CBF', $fixing);
+        }
+        if (PHP_CODESNIFFER_CBF !== $fixing) {
+            throw new LogicException('PHP_CODESNIFFER_CBF was defined with an incorrect value');
+        }
+
+        $sniffer = new Runner();
+        $sniffer->checkRequirements();
+        $sniffer->config = $this->prepareConfig($usingExplicitStandard, $lintLanguage, $fixing);
+        $sniffer->init();
+        $sniffer->reporter = new Reporter($sniffer->config);
+
+        // Find all the relevant code blocks for linting
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            echo 'Finding markdown files... ';
+        }
+
+        $files = new FileList($sniffer->config, $sniffer->ruleset);
+
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            $numFiles = count($files);
+            echo "DONE ($numFiles files in queue)" . PHP_EOL;
+        }
+
+        $codeBlocks = $this->findFencedCodeblocks($files, $lintLanguage);
+
+        // Add code blocks to the file list for linting
+        $todo = [];
+        foreach ($codeBlocks as $block) {
+            $dummy = new CodeBlock($block['content'], $sniffer->ruleset, $sniffer->config);
+            $dummy->num = $block['num'];
+            $dummy->path = $block['path'];
+            $dummy->realPath = $block['realpath'];
+            $todo[] = $dummy;
+        }
+
+        // Do the actual linting
+        $numErrors = $this->sniff($sniffer, $todo);
+
+        // The dummy files have the fixed content stored - but we still need to write that to the original files.
+        // There's no good AST to markdown renderer for league/commonmark so we're just doing a bit of an ugly
+        // search and replace.
+        if ($fixing) {
+            /** @var CodeBlock $dummy */
+            foreach ($todo as $dummy) {
+                if ($dummy->getFixedCount() < 1) {
+                    continue;
+                }
+
+                if (!is_file($dummy->realPath)) {
+                    // 3 is the exit code phpcs uses for errors like this
+                    throw new DeepExitException("Can't find file {$dummy->realPath} to set new content", 3);
+                }
+
+                /** @var FencedCode $mdBlock */
+                $mdBlock = $codeBlocks[$dummy->path]['md'];
+                $indent = str_repeat(' ', $mdBlock->getOffset());
+                $origBlockContent = preg_replace('/^/m', $indent, $mdBlock->getLiteral());
+                $newBlockContent = preg_replace('/^/m', $indent, preg_replace('/\s*<\?php\n?/', '', $dummy->getContent()));
+                $newFileContent = str_replace($origBlockContent, $newBlockContent, file_get_contents($dummy->realPath));
```

Comment:
```suggestion
                $contents = file_get_contents($dummy->realPath);
                $newFileContent = str_replace($origBlockContent, $newBlockContent, $contents);
```

Also is $dummy->getContent() the same thing as file_get_contents($dummy->realPath);? If so should probably just use $dummy->getContent()

Reply:
`$dummy->getContent()` gets whatever the content is at the time of getting the content. If the block has been "fixed" then it will be different than the original file content.